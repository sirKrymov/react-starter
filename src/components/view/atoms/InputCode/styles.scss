@import '../../../../styles/configs';

.input-code {
  &__field {
    input {
      background-color: var(--bgPrimary);
      -moz-appearance: textfield;
      border-radius: $border-radius;
      margin-right: 10px;
      font-weight: 500;
      text-align: center;
      outline: none;
      border: 1px solid var(--borderPrimary);
      height: 38px;
      width: 50px;

      &:last-child {
        margin-right: 0;
      }
    }

    input::-webkit-outer-spin-button,
    input::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
    }
  }
}
