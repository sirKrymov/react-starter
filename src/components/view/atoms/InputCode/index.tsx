// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ChangeEvent, ReactElement } from 'react';

import ReactCodeInput from 'react-code-input';
import cn from 'classnames';

import './styles.scss';

interface IProps {
  onChange(e: ChangeEvent<any>): void;
  className?: string;
  disabled?: boolean;
  value: string;
  id: string;
}

export function InputCode({
  className,
  onChange,
  disabled,
  value,
  id
}: IProps): ReactElement {
  const inputCodeClass = cn(
    {
      'input-code': true
    },
    className
  );

  return (
    <div className={inputCodeClass}>
      <ReactCodeInput
        className="input-code__field"
        onChange={onChange}
        disabled={disabled}
        fields={4}
        value={value}
        type="number"
        id={id}
      />
    </div>
  );
}
