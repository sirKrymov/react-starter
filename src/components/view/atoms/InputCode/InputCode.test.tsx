import React from 'react';

import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { InputCode } from '../InputCode';

describe('<InputCode> compound component', () => {
  const handleChange = jest.fn();

  let component: ReactTestRenderer;
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    component = renderer.create(
      <InputCode onChange={handleChange} value="" id="check-input-phone" />
    );
  };

  const createShallowComponent = () => {
    shallowComponent = shallow(
      <InputCode onChange={handleChange} value="" id="check-input-phone" />
    );
  };

  describe('WHEN the component loads', () => {
    it('THEN it should render without crashing', () => {
      createComponent();
      expect(component.toJSON()).toBeDefined();
    });

    it('THEN it should match snapshot', () => {
      createShallowComponent();
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
