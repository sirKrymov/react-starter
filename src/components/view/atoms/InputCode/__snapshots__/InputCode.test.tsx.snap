// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputCode> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="input-code"
>
  <t
    autoFocus={true}
    className="input-code__field"
    disabled={false}
    fields={4}
    filterChars={
      Array [
        "-",
        ".",
      ]
    }
    filterKeyCodes={
      Array [
        189,
        190,
      ]
    }
    forceUppercase={false}
    id="check-input-phone"
    isValid={true}
    onChange={[MockFunction]}
    type="number"
    value=""
  />
</div>
`;
